### PR TITLE
Improvements around streams

### PR DESCRIPTION
**Pull request by Bond-009**
* Use ArrayPool instead of allocating new buffers each time
* Remove NetworkStream copy
* Remove some dead code

### CODE REVIEW

File: Emby.Server.Implementations/Net/UdpSocket.cs [hunk 1]
```diff
@@ -236,35 +233,40 @@ public IAsyncResult BeginSendTo(byte[] buffer, int offset, int size, IpEndPointI
...
         public int EndSendTo(IAsyncResult result)
         {
             ThrowIfDisposed();
 
-            return _Socket.EndSendTo(result);
+            return _socket.EndSendTo(result);
         }
 
-        protected override void Dispose(bool disposing)
+        private void ThrowIfDisposed()
```
Comment (JustAMan) [inline Emby.Server.Implementations/Net/UdpSocket.cs]: who's going to call this function?

File: MediaBrowser.Api/LiveTv/ProgressiveFileCopier.cs [hunk 2]
```diff
@@ -63,28 +47,23 @@ public async Task WriteToAsync(Stream outputStream, CancellationToken cancellati
...
-                if (StartPosition > 0)
-                {
-                    inputStream.Position = StartPosition;
-                }
+                fileOptions |= FileOptions.Asynchronous;
+            }
 
+            using (var inputStream = new FileStream(_path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite, 4096, fileOptions))
+            {
                 var emptyReadLimit = AllowEndOfFile ? 20 : 100;
```
Comment (JustAMan) [inline MediaBrowser.Api/LiveTv/ProgressiveFileCopier.cs]: what is this insanity?..

File: Emby.Server.Implementations/Net/UdpSocket.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Net/UdpSocket.cs]: It's already being used by the internal functions.

File: MediaBrowser.Api/LiveTv/ProgressiveFileCopier.cs (hunk 2 above)
Comment (Bond-009) [reply MediaBrowser.Api/LiveTv/ProgressiveFileCopier.cs]: 🤷‍♂️ It's bad...

**Pull request merged**